lib/exercism/user.rb

```diff
@@ -80,6 +90,10 @@ def self.find_by_username(username)
     find_by(username: username)
   end
 
+  def self.find_by_username_and_share_key(username, key)
+    find_by(username: username, share_key: key)
+  end
```

Comment:
I think you can directly use `find_by` instead of creating this method. Such dynamic finders were [removed from ActiveRecord (4.1).](https://github.com/rails/activerecord-deprecated_finders)